src/hotspot/share/utilities/vmError.cpp

```diff
@@ -166,6 +170,48 @@ static void print_bug_submit_message(outputStream *out, Thread *thread) {
   out->print_raw_cr("#");
 }
 
+static bool check_stack_headroom(Thread* thread,
+                                 size_t headroom) {
+  static const address stack_top    = thread != nullptr
+                                    ? thread->stack_base()
+                                    : os::current_stack_base();
+  static const size_t  stack_size   = thread != nullptr
+                                    ? thread->stack_size()
+                                    : os::current_stack_size();
```

Comment:
Why even bother with thread? If you want this to work without Thread*, you may just as well just use the os::current_stack_xxx() functions. 

OTOH I think it would also be perfectly acceptable to just use Thread*, since we have that in 99% of cases, and if Thread is null to assume that we have enough headroom.

Reply:
I also would not bother making these vars static. Your intent was optimization, right, since we should only call this for one thread? But its surprising in the case we ever want to call this from different threads.